selftests/unit/test_jobdata.py

```diff
@@ -14,78 +14,65 @@ class JobdataTest(unittest.TestCase):
-                                 ["failfast", False, None],
-                                 ["ignore_missing_references", False, None],
-                                 ["execution_order", "variants-per-test",
-                                  None]]:
-                    act = args.get(scenario[0])
-                    for exp in scenario[1:]:
-                        if act == exp:
-                            break
-                    else:
-                        errs.append("args: Invalid value '%s' of key '%s' "
-                                    "%s" % (act, scenario[0],
-                                            scenario[1:]))
-            else:
-                errs.append("args: Invalid args: %s" % args)
+            self.fail("args: Unable to retrieve: %s" % details)
+        self.assertTrue(isinstance(args, dict),
+                        "args: Invalid args: %s" % args)
+        for scenario in [["loaders", [u"external:/bin/echo"]],
+                         ["external_runner", u"/bin/echo"],
+                         ["failfast", False, None],
+                         ["ignore_missing_references", False, None],
+                         ["execution_order", "variants-per-test",
+                          None]]:
+            act = args.get(scenario[0])
+            self.assertIn(act, scenario[1:],
+                          "args: Invalid value '%s' of key '%s' '%s'" % (
+                              act, scenario[0], scenario[1:]))
 
         # config
         conf_path = jobdata.retrieve_config(pth)
-        if os.path.exists(conf_path):
-            exp = "[avocado.selftest]\njobdata = yes"
-            with open(conf_path, "r") as conf:
-                act = conf.read()
-                if exp not in act:
-                    errs.append("config: Expected string\n%s\n\nNot in:\n%s"
-                                % (exp, act))
-
-        else:
-            errs.append("config: Retrieved path '%s' does not exists"
-                        % conf_path)
+        self.assertTrue(os.path.exists(conf_path),
+                        "config: Retrieved path '%s' does not exists" %
+                        conf_path)
+        exp = "[avocado.selftest]\njobdata = yes"
+        with open(conf_path, "r") as conf:
+            act = conf.read()
+            self.assertIn(exp, act,
+                          "config: Expected string\n%s\n\nNot in:\n%s" % (
+                              exp, act))
 
         # cmdline
         act = jobdata.retrieve_cmdline(pth)
         exp = ['/usr/local/bin/avocado', 'run', '--external-runner',
                '/bin/echo', '-m', 'examples/mux-0.yaml', '--', 'yes', 'no']
-        if exp != act:
-            errs.append("cmdline: Invalid cmdline '%s' (%s)" % (act, exp))
-        self.assertFalse(errs, "Errors: %s" % "\n  ".join(errs))
+        self.assertEqual(exp, act,
+                         "cmdline: Invalid cmdline '%s' (%s)" % (act, exp))
```

Comment:
ok, I don't see changes in the coverage and the code looks good. Will just run some manual tests before merging.